src/Accelerators/NNPA/Dialect/ZHigh/ZHighOps/OpHelper.cpp

```diff
@@ -76,7 +76,9 @@ ZTensorEncodingAttr::DataLayout getZTensorDataLayoutByRank(int64_t rank) {
   else if (rank == 2)
     return ZTensorEncodingAttr::DataLayout::_2D;
   else if (rank == 3)
-    return ZTensorEncodingAttr::DataLayout::_3D;
+    // Use 3DS instead of 3D since important ops like LSTM/MatMul/Softmax use
+    // 3DS, which reduces the number of layout transformations.
+    return ZTensorEncodingAttr::DataLayout::_3DS;
```

Comment:
Unrelated, it would be useful to define each of the layouts; maybe where they are expanded as maps (`convertZTensorToMemRefType`)? It would be nice to have one spot where a bit more info is given about these.

Reply:
@AlexandreEichenberger I checked with Resnet50-v1-12, there is no difference in running time. Conv-based networks use NHWC, and its activation, such as, Relu, correctly use NHWC via our layout propagation.